projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-jwt.php

```diff
@@ -193,12 +193,12 @@ public static function encode( $payload, $key, $alg = 'HS256', $key_id = null, $
 		}
 
 		$segments      = array();
-		$segments[]    = static::urlsafe_b64_decode( static::json_encode( $header ) );
-		$segments[]    = static::urlsafe_b64_decode( static::json_encode( $payload ) );
+		$segments[]    = static::urlsafe_b64_encode( static::json_encode( $header ) );
```

Comment:
It is only used to decode and not to encode...
I foud the issue by replacing with [the WPCOM class](fbhepr%2Skers%2Sgehax%2Sjc%2Qpbagrag%2Syvo%2Swjg%2Spynff.wjg.cuc%23-og) and inspecting the differences (it is mainly a lot of lint changes), adding a new cypher algorythm AND using encode here instead of decode...

Have a look [here](fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Swjg%2Spynff.wjg.cuc%3Se%3Qpn0osq59%23159%2Q165-og), it is obvious.

Reply:
This is all a nice find.
Can you please create a release note for that?

Reply:
Done